youtube_dlc/downloader/hls.py

```diff
@@ -211,6 +222,14 @@ def is_ad_fragment_end(s):
                 elif is_ad_fragment_end(line):
                     ad_frag_next = False
 
+        if external_downloader == 'aria2c':
+            ctx['url_list'] = fragment_urls
+            ctx['decrypt_info'] = decrypt_info
+            self.urlopen = self.ydl.urlopen
+
+            downloader = Aria2cFD(self.ydl, self.params)
+            downloader.real_download(ctx['filename'], ctx)
```

Comment:
Similar to dash, failure should be handled

Reply:
It's handled now.